### ISSUE TITLE
python-lsp-server

### ISSUE THREAD

**Issue by Mo8it** [SmiteshP/nvim-navic#2]
Thanks for this plugin! I used nvim-gps before, but I needed it to be working with Julia. This plugin works with Julia, but not with Python.

I get the following error while using [python-lsp-server](https://github.com/python-lsp/python-lsp-server):

```
Error executing vim.schedule lua callback: ...ite/pack/packer/start/nvim-navic/lua/nvim-navic/init.lua:30: attempt to index l
ocal 'scope' (a nil value)                                                                                                   
stack traceback:                                                                                                             
        ...ite/pack/packer/start/nvim-navic/lua/nvim-navic/init.lua:30: in function 'dfs'                                    
        ...ite/pack/packer/start/nvim-navic/lua/nvim-navic/init.lua:50: in function 'parse'                                  
        ...ite/pack/packer/start/nvim-navic/lua/nvim-navic/init.lua:59: in function 'handler'                                
        ...ite/pack/packer/start/nvim-navic/lua/nvim-navic/init.lua:13: in function 'callback'                               
        /usr/share/nvim/runtime/lua/vim/lsp.lua:1535: in function 'handler'                                                  
        /usr/share/nvim/runtime/lua/vim/lsp.lua:1025: in function ''                                                         
        vim/_editor.lua: in function <vim/_editor.lua:0>
```

**Comment by SmiteshP** [comment]
Tried this out with some dummy python code

```python
class myclass:

    def mymethod(self):
        print("hello world!")
```

The data returned by the python-lsp-server was the following:
```
{ {
    kind = 5,
    location = {
      range = {
        end = {
          character = 29,
          line = 3
        },
        start = {
          character = 0,
          line = 0
        }
      },
      uri = "file:///home/smith/Desktop/temp.py"
    },
    name = "myclass"
  }, {
    containerName = "myclass",
    kind = 6,
    location = {
      range = {
        end = {
          character = 29,
          line = 3
        },
        start = {
          character = 4,
          line = 2
        }
      },
      uri = "file:///home/smith/Desktop/temp.py"
    },
    name = "mymethod"
  } }
```

Whereas data returned by pyright for the same file was 
```
{ {
    children = { {
        children = {},
        kind = 6,
        name = "mymethod",
        range = {
          end = {
            character = 29,
            line = 3
          },
          start = {
            character = 4,
            line = 2
          }
        },
        selectionRange = {
          end = {
            character = 16,
            line = 2
          },
          start = {
            character = 8,
            line = 2
          }
        }
      } },
    kind = 5,
    name = "myclass",
    range = {
      end = {
        character = 29,
        line = 3
      },
      start = {
        character = 0,
        line = 0
      }
    },
    selectionRange = {
      end = {
        character = 13,
        line = 0
      },
      start = {
        character = 6,
        line = 0
      }
    }
  } }
```

So it seems that pylsp is using the older specification and returning [SymbolInformation](https://microsoft.github.io/language-server-protocol/specifications/lsp/3.17/specification/#symbolInformation). I have a check for this in my `attach` function at https://github.com/SmiteshP/nvim-navic/blob/master/lua/nvim-navic/init.lua#L310-L313.

So in conclusion, pylsp is returning `true` for documentSymbol server capabilities. But instead returns the older SymbolInformation struct. This is a bug from pylsp side I believe. It should either return false in server capabilities for documentSymbol or return the documentSymbol as expected.

**Comment by SmiteshP** [comment]
@Mo8it I see that there are already issue on their repo for supporting documentSymbols. So probably they are working on it.
It can't be solved by navic 😞 Its just not possible to extract hierarchical context from the information it provides right now.

**Comment by SmiteshP** [comment]
Or you could give pyright a try, its been working great for me.

**Comment by Mo8it** [comment]
OK, thanks a lot for the analysis and effort!

**Comment by Mo8it** [comment]
Small update:
Switching from pylsp to https://github.com/pappasam/jedi-language-server did solve my problem :)
